### PR TITLE
Convert parser input interface to api::Term

### PR DESCRIPTION
**Pull request by ajreynol**
In preparation for parser migration.

This PR updates the Input classes to return api::Term instead of Expr.

It patches the nra_examples but does not address https://github.com/CVC4/cvc4-projects/issues/108.

### CODE REVIEW

File: src/parser/parser.cpp [hunk 1]
```diff
@@ -661,14 +661,14 @@ Command* Parser::nextCommand()
...
-Expr Parser::nextExpression()
+api::Term Parser::nextExpression()
 {
   Debug("parser") << "nextExpression()" << std::endl;
   d_resourceManager->spendResource(ResourceManager::Resource::ParseStep);
   Expr result;
   if (!done()) {
     try {
-      result = d_input->parseExpr();
+      result = d_input->parseExpr().getExpr();
```
Comment (4tXJ7f) [inline src/parser/parser.cpp]: Why not just make the `result` a `Term`?

Comment (ajreynol) [reply src/parser/parser.cpp]: It is on my branch, this change touches fewer lines (will have fewer merge conflicts this way).

Comment (ajreynol) [reply src/parser/parser.cpp]: Actually i guess its even 2 vs 2 haha. Anyways that was the reason.

Comment (ajreynol) [reply src/parser/parser.cpp]: Actually nevermind, this way has fewer merge conflicts :)  Anyways that was supposely the reason.

File: src/parser/parser.cpp [hunk 2]
```diff
@@ -665,10 +665,10 @@ api::Term Parser::nextExpression()
 {
   Debug("parser") << "nextExpression()" << std::endl;
   d_resourceManager->spendResource(ResourceManager::Resource::ParseStep);
-  Expr result;
+  api::Term result;
   if (!done()) {
     try {
-      result = d_input->parseExpr().getExpr();
+      result = d_input->parseExpr();
```
Comment (4tXJ7f) [inline src/parser/parser.cpp]: The `return api::Term(result);` statement below can be replaced by just `return result;`, I think.

**Pull request merged**